library/src/main/scala/scala/scalajs/LinkingInfo.scala

```diff
@@ -108,4 +108,35 @@ object LinkingInfo {
   def assumingES6: Boolean =
     linkingInfo.assumingES6
 
+  /** The value of the global JavaScript `this`.
+   *
+   *  This returns the value that would be obtained by writing `this` at the
+   *  top-level of the JavaScript file generated by Scala.js. In most
+   *  JavaScript environments, this is equivalent to the *global object*, but
+   *  it is not necessarily the case. For example, on Node.js, `this` is the
+   *  object representing the `exports` of the current module.
+   *
+   *  Using this value should be rare, and mostly limited to writing code
+   *  detecting what the global object is. For example, a typical detection
+   *  code looks like:
+   *  {{{
+   *  val globalObject = {
+   *    import js.Dynamic.{global => g}
+   *    if (js.typeOf(g.global) != "undefined" && (g.global.Object eq g.Object)) {
+   *      // Node.js environment detected
+   *      g.global
+   *    } else {
+   *      // In all other well-known environment, we can use the global `this`
+   *      scala.scalajs.LinkingInfo.globalThis
+   *    }
+   *  }
+   *  }}}
+   *  Note that the above code is not comprehensive, as there can be JavaScript
+   *  environments where the global object cannot be fetched neither through
+   *  `global` nor `this`. If your code needs to run in such an environment, it
+   *  is up to you to use an appropriate detection procedure.
+   */
+  @inline
+  def globalThis: js.Dynamic = linkingInfo.globalThis
+
```

Comment:
> Do you object to it being defined in `scalajs.runtime.linkingInfo.globalThis` too?

No, that's fine. Implementation-wise, `globalThis` can be argued to satisfy the conditions for being inside `LinkingInfo` :P

> I have secret plans to kill envInfo altogether. 

I don't fully understand this. AFAICS, there is currently no public `EnvInfo`, only a facade type for documentation (that does not even appear in the doc anymore).

>  but I don't like it because it gives too much "visibility" to that method.

I thought this as well for quite a while, but is it really that bad?

As I am thinking of this, I start wondering: Should `global` itself not be on `js` itself? I would argue that no, because it makes it clearer that code is dynamically typed. Compare:

```scala
js.Dynamic.global.foo(a, b, c)  // oh, foo is dynamic!
js.global.foo(a, b, c)          // it's probably fine if it compiles (!)
js.Promise.all(a, b, c)         // it **is** fine if it compiles.
```

The same argument can be made for `globalThis` of course (and for `constructorOf`, but I recall that there were plans to type it better, though didn't find anything).

There is another option I just thought of: We could start an package `js.completeness` (or something) where we put features we think should rarely / never be used but we provide for completeness. Upside: It is *very* obvious that we don't want people to use it. Downside: I don't think there is anything else upcoming we'll put in there.